src/openms/include/OpenMS/FORMAT/MzTabM.h

```diff
@@ -0,0 +1,320 @@
+  {
+  public:
+    /// Default constructor
+    MzTabM() = default;
+
+    /// Destructor
+    ~MzTabM() = default;
+
+    const MzTabMMetaData& getMetaData() const;
+
+    void setMetaData(const MzTabMMetaData& m_md);
+
+    const MzTabMSmallMoleculeSectionRows& getMSmallMoleculeSectionRows() const;
+
+    void setMSmallMoleculeSectionRows(const MzTabMSmallMoleculeSectionRows& m_smsd);
+
+    const MzTabMSmallMoleculeFeatureSectionRows& getMSmallMoleculeFeatureSectionRows() const;
+
+    void setMSmallMoleculeFeatureSectionRows(const MzTabMSmallMoleculeFeatureSectionRows& m_smfsd);
+
+    const MzTabMSmallMoleculeEvidenceSectionRows& getMSmallMoleculeEvidenceSectionRows() const;
+
+    void setMSmallMoleculeEvidenceSectionRows(const MzTabMSmallMoleculeEvidenceSectionRows& m_smesd);
+
+    void setCommentRows(const std::map<Size, String>& com);
+
+    void setEmptyRows(const std::vector<Size>& empty);
+
+    const std::vector<Size>& getEmptyRows() const;
+
+    const std::map<Size, String>& getCommentRows() const;
+
+    /// Extract opt_ (custom, optional column names)
+    std::vector<String> getMSmallMoleculeOptionalColumnNames() const;
+
+    /// Extract opt_ (custom, optional column names)
+    std::vector<String> getMSmallMoleculeFeatureOptionalColumnNames() const;
+
+    /// Extract opt_ (custom, optional column names)
+    std::vector<String> getMSmallMoleculeEvidenceOptionalColumnNames() const;
+
+    static void addMetaInfoToOptionalColumns(const std::set<String>& keys,
+                                             std::vector<MzTabOptionalColumnEntry>& opt,
+                                             const String& id, const MetaInfoInterface& meta);
+
+  /**
+  * @brief Export FeatureMap with Identifications to MzTabM
+  *
+  * @return MzTabM object
+  */
+  static MzTabM exportFeatureMapToMzTabM(const FeatureMap& feature_map);
+
+  protected:
+
+    MzTabMMetaData m_meta_data_;
+    MzTabMSmallMoleculeSectionRows m_small_molecule_data_;
+    MzTabMSmallMoleculeFeatureSectionRows m_small_molecule_feature_data_;
+    MzTabMSmallMoleculeEvidenceSectionRows m_small_molecule_evidence_data_;
+    std::vector<Size> empty_rows_; ///< index of empty rows
+    std::map<Size, String> comment_rows_; ///< comments
+    std::vector<String> sml_optional_column_names_;
+    std::vector<String> smf_optional_column_names_;
+    std::vector<String> sme_optional_column_names_;
+
+    static String getAdductString_(const IdentificationDataInternal::ObservationMatchRef& match_ref);
+
+    static void getFeatureMapMetaValues_(const FeatureMap& feature_map,
+                                         std::set<String>& feature_user_value_keys,
+                                         std::set<String>& observationmatch_user_value_keys,
+                                         std::set<String>& compound_user_value_keys);
+
+    template <class ForwardIterator>
+    static void replaceWhiteSpaces_(ForwardIterator first, ForwardIterator last)
```

Comment:
why is this needed? do the standard stl algorithms or our String helper don't work?